### PR TITLE
Stats: Adding skip button functionality

### PR DESCRIPTION
**Pull request by grzegorz-cp**
Related to #86136

## Proposed Changes

* adding the following skip button functionality
   * adding a mutation that updates `focus-jetpack-purchase`
   * triggering an event before redirecting to the main page
   * redirecting to the main page

## Testing Instructions

* create a JN blog and connect Jetpack
* open the live branch and navigate to the Stats page for the new blog
* apply `stats/checkout-flows-v2` feature flag and verify that you are redirected to a purchase page
* click on `I'll do it later` button and verify in the network tab that a call was made to update the nieces and a Tracks event was triggered
* when you navigate or you are redirected to the `Traffic` page you will still end up on the purchase page. This is due to react-query caching and will be addressed after #86625 is merged

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by a8ck3n — APPROVED** [review]
Tests well for me. I'm seeing the Tracks event and the call to update the notice values in the Network dev tools. On my first pass through, clicking redirected me to the commercial purchase page. On my second and subsequent passes, it loaded the Stats page without a redirect though. I was expecting to still be redirected to the purchase page. Does that sound correct @grzegorz-cp?

File: client/my-sites/stats/stats-purchase/stats-purchase-personal.tsx [hunk 1]
```diff
@@ -85,33 +90,35 @@ const PersonalPurchase = ( {
...
 		: translate( 'Get Jetpack Stats' );
+
 	if ( isNewPurchaseFlowEnabled ) {
 		continueButtonText = translate( 'Contribute and continue' );
 	}
 
+	const { mutateAsync: mutateNoticeVisbilityAsync } = useNoticeVisibilityMutation(
+		siteId,
+		'focus_jetpack_purchase'
+	);
```
Comment (dognose24) [inline client/my-sites/stats/stats-purchase/stats-purchase-personal.tsx:110]: Do we plan to skip the purchase page permanently or postpone it for a while?

Comment (kangzj) [reply client/my-sites/stats/stats-purchase/stats-purchase-personal.tsx:110]: I don't think we defined this anywhere - I'm thinking maybe we only snooze it for like a month?

**Comment by grzegorz-cp** [comment]
> Tests well for me. I'm seeing the Tracks event and the call to update the notice values in the Network dev tools. On my first pass through, clicking redirected me to the commercial purchase page. On my second and subsequent passes, it loaded the Stats page without a redirect though. I was expecting to still be redirected to the purchase page. Does that sound correct @grzegorz-cp?

After you click the skip button, you should be redirected to the Traffic page, but it's inconsistent in this PR, and I'll adjust it once #86625 is merged.

File: client/my-sites/stats/stats-purchase/stats-purchase-personal.tsx (hunk 1 above)
Comment (grzegorz-cp) [reply client/my-sites/stats/stats-purchase/stats-purchase-personal.tsx:110]: Ok, I'll add a ticket for that and we and apply the same logic we used for the usage notice.

**Pull request merged**